modules/services/gpg-agent.nix

```diff
@@ -22,6 +22,16 @@ let
     set -gx GPG_TTY (tty)
   '' + optionalString cfg.enableSshSupport gpgSshSupportStr;
 
+  gpgNushellInitStr = ''
+    $env.GPG_TTY = (tty)
+  '' + optionalString cfg.enableSshSupport ''
+    ${gpgPkg}/bin/gpg-connect-agent updatestartuptty /bye | ignore
+
+    if not "SSH_AUTH_SOCK" in $env {
+      $env.SSH_AUTH_SOCK = (${gpgPkg}/bin/gpgconf --list-dirs agent-ssh-socket)
+    }
```

Comment:
if we can find a way to move this to `home.sessionVariables` then I can remove this block but I can confirm (on my machine) that this works here.